pyccel/ast/core.py

```diff
@@ -174,13 +175,77 @@
 }
 
 #==============================================================================
-# Pow, Add, Mul need to inherite sympy.Boolean to be able to use them in a logical expression
+def broadcast(shape_1, shape_2):
+    """ This function broadcast two shapes using numpy broadcasting rules """
+    a = len(shape_1)
+    b = len(shape_2)
+    if a>b:
+        new_shape_2 = (1,)*(a-b) + tuple(shape_2)
+        new_shape_1 = shape_2
+    elif b>a:
+        new_shape_1 = (1,)*(b-a) + tuple(shape_1)
+        new_shape_2 = shape_2
+    else:
+        new_shape_2 = shape_2
+        new_shape_1 = shape_1
+    
+    new_shape = []
+    for e1,e2 in zip(new_shape_1, new_shape_2):
+        if e1 == e2:
+            new_shape.append(e1)
+        elif e1 == 1:
+            new_shape.append(e2)
+        elif e2 == 2:
+            new_shape.append(e1)
+        else:
+            msg = 'operands could not be broadcast together with shapes {} {}'
+            msg = msg.format(shape_1, shape_2)
+            raise ValueError(msg)
+    return tuple(new_shape)
```

Comment:
Is this the same function as in pyccel/ast/builtins.py? If so why is it copied and pasted?

Reply:
yes it's the same ,now there is only one copy